### ISSUE TITLE
Leased version of ExecuteAsync

### ISSUE THREAD

**Issue by prat0088** [StackExchange/StackExchange.Redis#1257]
StringGetLease is great, but I can't use it for Redis Modules and custom commands.  My specific use case is that I need to return a large string (actually a byte array) from Redis to parse.

I'm not very familiar with the deep internals of StackExchange.Redis, but I'm wondering how big of a change this will be.  Would it be enough to add a new class, ScriptResultLease, that is essentially the same but passes result.AsLease() to SetResult() instead of just result as it does [here](https://github.com/StackExchange/StackExchange.Redis/blob/8672ca23b453921cb26c821375bef38d87e97b63/src/StackExchange.Redis/ResultProcessor.cs#L1371)?  Then add ExecuteAsyncLease() with ResultProcessor.ScriptResultLease around [here](https://github.com/StackExchange/StackExchange.Redis/blob/master/src/StackExchange.Redis/RedisDatabase.cs#L1167).

Basically I'm wondering how to efficiently read a large byte array from a redis module command without any additional allocations, and if I can't do that today, how big of a change it would require.

**Comment by NickCraver** [comment]
Tagging @mgravell here for API thoughts.
